jf_agent/agent_logging.py

```diff
@@ -217,6 +219,53 @@ def filter(self, record: LogRecord) -> bool:
         return not record.__dict__.get(AGENT_LOG_TAG, False)
 
 
+def configure_structlog() -> None:
+    """
+    Configures rendering of log messages within structlog, which then passes a fully formatted string to
+    the logging module for output.
+    """
+    structlog.stdlib.recreate_defaults()
+    structlog.configure(
+        processors=[
+            structlog.contextvars.merge_contextvars,
+            structlog.stdlib.filter_by_level,
+            structlog.stdlib.add_logger_name,
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            structlog.processors.TimeStamper(fmt="iso"),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.processors.UnicodeDecoder(),
+            structlog.processors.CallsiteParameterAdder(
+                {
+                    structlog.processors.CallsiteParameter.FILENAME,
+                    structlog.processors.CallsiteParameter.FUNC_NAME,
+                    structlog.processors.CallsiteParameter.LINENO,
+                }
+            ),
+            structlog.dev.ConsoleRenderer(),
+        ],
+        wrapper_class=structlog.stdlib.BoundLogger,
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        cache_logger_on_first_use=True,
+    )
+
+
+def bind_context_vars(
```

Comment:
Nit: This is somewhat confusingly similar to the structlog `bind_contextvars`, but with different functionality. Might change to something like `bind_default_agent_context` or similar

Reply:
Good point, I can update the function name